### PR TITLE
Re-skeletonize

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request re-skeletonizes this repository.

## 💭 Motivation and context ##

This repository had been started from a skeleton in the distant past, but it was woefully out of date and lacked even a Lineage configuration.

## 🧪 Testing ##

All relevant automated testing passes.  The JavaScript LGTM check fails because there is no JavaScript code in this repository.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    ignore:
+      - dependency-name: "ansible"
+      - dependency-name: "ansible-lint"
+
+  - package-ecosystem: "terraform"
+    directory: "/terraform"
+    schedule:
+      interval: "weekly"
+    # ignore:
+    #   - dependency-name: "hashicorp/aws"
```
Comment (dav3r) [inline .github/dependabot.yml]: We want to uncomment this bit inherited from `skeleton-ansible-role-with-test-user`:
```suggestion
    ignore:
      - dependency-name: "hashicorp/aws"
```

File: README.md [hunk 2]
```diff
@@ -7,18 +7,66 @@
...
-  Debian package file.  Default value: `"cisa-cool-third-party-production"`
-- `package_file` - The name of the Nessus Debian package file.
-  Default value: `"Nessus-{{ version }}-debian6_amd64.deb"`
-- `version` - The version number of the Nessus Debian package file.
-  Default value: `"8.7.1"`
+| Variable | Description | Default | Required |
+|----------|-------------|---------|----------|
+| package_bucket | The name of the AWS S3 bucket containing the Nessus Debian package file. | `cisa-cool-third-party-production` | No |
+| package_file | The name of the Nessus Debian package file. | `Nessus-{{ version }}-debian6_amd64.deb` | No |
+| version | The version number of the Nessus Debian package file. | `8.7.1` | No |
```
Comment (dav3r) [inline README.md:69]: I was speaking with @mcdonnnj and he mentioned that he thought you would be updating us to the newest version of Nessus 8.x (currently `8.15.4`).  I just wanted to confirm if you planned to do that in a separate PR or if you were going to include it in this one.

File: terraform/providers.tf [hunk 3]
```diff
@@ -3,37 +3,77 @@
...
   region = var.aws_region
 }
 
-# The provider used to create roles that can read objects from the
-# production COOL "third-party" bucket
+# The provider used to create the role that can be assumed to do
+# everything the CI user needs to do in the staging account.
 provider "aws" {
-  alias  = "images-production"
+  alias = "images_staging_provisionaccount"
```
Comment (dav3r) [inline terraform/providers.tf]: Can you please re-order these providers alphabetically so that `aws.images_production_provisionaccount` comes before `aws.images_staging_provisionaccount`?

File: terraform/variables.tf [hunk 4]
```diff
@@ -30,7 +31,7 @@ variable "tags" {
   description = "Tags to apply to all AWS resources created"
 
   default = {
-    Team        = "NCATS OIS - Development"
-    Application = "ansible-role-nessus testing"
+    Team        = "VM Fusion - Development"
+    Application = "skeleton-ansible-role-with-test-user testing"
```
Comment (dav3r) [inline terraform/variables.tf]: Looks like you accidentally overwrote this:
```suggestion
    Application = "ansible-role-nessus testing"
```

**Review by dav3r — CHANGES_REQUESTED** [review]
Good stuff, though I spotted a few thangs you missed.

Thanks for being a re-skeletonizer! 💀

File: README.md (hunk 2 above)
Comment (jsf9k) [reply README.md:69]: I created #16, which failed automated testing; that's how I realized that I needed to re-skeletonize.

File: terraform/providers.tf (hunk 3 above)
Comment (jsf9k) [reply terraform/providers.tf]: Please see commit 99bfd75.

File: README.md (hunk 2 above)
Comment (dav3r) [reply README.md:69]: Thanks- I should've checked myself (hence why I wrecked myself).

**Review by dav3r — APPROVED** [review]
Outstanding! 🤖

File: meta/main.yml [hunk 5]
```diff
@@ -3,18 +3,26 @@ galaxy_info:
...
         - buster
+        - bullseye
         # Kali linux isn't an option here, but it is based on
         # Debian Testing:
         # https://www.kali.org/docs/policy/kali-linux-relationship-with-debian
-        - bullseye
-  galaxy_tags:
-    - nessus
+        - bookworm
+  role_name: nessus
```
Comment (mcdonnnj) [inline meta/main.yml:30]: Since these platforms are in the molecule configuration now.

```suggestion
    - name: Ubuntu
      versions:
        - bionic
        - focal
  role_name: nessus
```

File: molecule/default/prepare.yml [hunk 6]
```diff
@@ -1,10 +1,15 @@
...
-- name: Prepare
+- name: Import upgrade playbook
+  ansible.builtin.import_playbook: upgrade.yml
+
+- name: Import python playbook
+  ansible.builtin.import_playbook: python.yml
+
+# boto3 is pre-installed anywhere we would be applying this Ansible
+# role
+- name: Install boto3 and pip
```
Comment (mcdonnnj) [inline molecule/default/prepare.yml]: Since `pip` is installed in the `python.yml` playbook above.

```suggestion
- name: Install boto3
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Thanks for tackling another one of these lost lambs. In addition to my suggestions below I believe we can now remove the `regex_replace()` logic in the following:
https://github.com/cisagov/ansible-role-nessus/blob/99bfd7572032a2d7ccefd612df19d7a72b4bc89e/tasks/main.yml#L52-L57

Here's a reference to a commit doing so in another Ansible role: https://github.com/cisagov/ansible-role-clamav/commit/8b0fb1792011089b013cb4add3f31de40be8cd68

**Comment by jsf9k** [comment]
> Thanks for tackling another one of these lost lambs. In addition to my suggestions below I believe we can now remove the `regex_replace()` logic in the following:
> 
> https://github.com/cisagov/ansible-role-nessus/blob/99bfd7572032a2d7ccefd612df19d7a72b4bc89e/tasks/main.yml#L52-L57
> 
> Here's a reference to a commit doing so in another Ansible role: [cisagov/ansible-role-clamav@8b0fb17](https://github.com/cisagov/ansible-role-clamav/commit/8b0fb1792011089b013cb4add3f31de40be8cd68)

Good call!

**Review by mcdonnnj — APPROVED** [review]
:shipit:
